probe-rs/src/architecture/arm/memory/adi_v5_memory_interface.rs

```diff
@@ -30,10 +31,13 @@ impl ADIMemoryInterface<ArmCommunicationInterface> {
         interface: ArmCommunicationInterface,
         access_port_number: impl Into<MemoryAP>,
     ) -> Self {
-        Self {
+        let mut interface = Self {
             interface,
             access_port: access_port_number.into(),
-        }
+            only_32bit_data_size: true,
+        };
+        interface.detect_data_size().ok(); // Ignore errors
+        interface
```

Comment:
I don't like this, but it simplifies things a lot

Reply:
What is so bad about returning an error? we return errors everywhere anyways?

Reply:
I tried to preserve current API. I will fix this in a minute.